repos/system_upgrade/el7toel8/actors/satelliteupgradefacts/actor.py

```diff
@@ -0,0 +1,133 @@
+            for nonscl_path in ['/var/lib/pgsql/data/', '/var/lib/pgsql/', '/var/lib/', '/']:
+                if os.path.exists(nonscl_path):
+                    nonscl_psql_stat = os.stat(nonscl_path)
+                    break
+
+            title = "Satellite PostgreSQL data migration"
+            flags = []
+            severity = reporting.Severity.MEDIUM
+
+            if scl_psql_stat.st_dev != nonscl_psql_stat.st_dev:
+                # get the current disk usage of the PostgreSQL data
+                scl_du_call = run(['du', '--block-size=1', '--summarize', scl_psql_path])
+                scl_used = int(scl_du_call['stdout'].split()[0])
+                # get the current free space on the target partition
+                nonscl_stat = os.statvfs(nonscl_path)
+                nonscl_free = nonscl_stat.f_bavail * nonscl_stat.f_frsize
+                if scl_used > nonscl_free:
+                    storage_message = "You currently don't have enough free storage to move the data. Automatic moving cannot be performed."
+                    flags = [reporting.Flags.INHIBITOR]
+                    severity = reporting.Severity.HIGH
+                else:
+                    storage_message = "You currently have enough free storage to move the data and this can be performed by the upgrade process."
+                summary = """
+                Your PostgreSQL data in {} is currently on a dedicated volume.
+                PostgreSQL on RHEL8 expects the data to live in /var/lib/pgsql/data.
+                {}
+                However, instead of moving the data over, you might want to consider manually adapting your mounts, so
+                that the contents of {} are available in /var/lib/pgsql/data.
+                """.format(scl_psql_path, storage_message, scl_psql_path)
+            else:
+                summary = "Your PostgreSQL data will be automatically migrated."
+
+            reporting.create_report([
+                reporting.Title(title),
+                reporting.Summary(textwrap.dedent(summary)),
+                reporting.Severity(severity),
+                reporting.Tags([]),
+                reporting.Flags(flags)
+            ])
+
+            self.produce(RpmTransactionTasks(
+                modules_to_enable=[Module(name='postgresql', stream='12')],
+                to_remove=['rh-postgresql12-runtime'],
+                to_install=['postgresql', 'postgresql-server']
+                )
+            )
+            if postgresql_contrib:
+                self.produce(RpmTransactionTasks(
+                    to_remove=['rh-postgresql12-postgresql-contrib'],
+                    to_install=['postgresql-contrib']
+                    )
+                )
+            if postgresql_evr:
+                self.produce(RpmTransactionTasks(
+                    to_remove=['rh-postgresql12-postgresql-evr'],
+                    to_install=['postgresql-evr']
+                    )
+                )
```

Comment:
All these RpmTransactionTasks can be combined into one big one, you'd just collect all the things to remove and add and make one big one out of it.